libs/fills/src/lib/fills.utils.ts

```diff
@@ -0,0 +1,163 @@
+import BigNumber from 'bignumber.js';
+import type {
+  FillFieldsFragment,
+  TradeFeeFieldsFragment,
+} from './__generated__/Fills';
+import * as Schema from '@vegaprotocol/types';
+
+export const TAKER = 'Taker';
+export const MAKER = 'Maker';
+
+export type Role = typeof TAKER | typeof MAKER | '-';
+
+export const getRoleAndFees = ({
+  data,
+  partyId,
+}: {
+  data: Pick<
+    FillFieldsFragment,
+    'buyerFee' | 'sellerFee' | 'buyer' | 'seller' | 'aggressor'
+  >;
+  partyId?: string;
+}): {
+  role: Role;
+  fees?: TradeFeeFieldsFragment;
+  marketState?: Schema.MarketState;
+} => {
+  let role: Role;
+  let fees;
+
+  if (data?.buyer.id === partyId) {
+    if (data.aggressor === Schema.Side.SIDE_BUY) {
+      role = TAKER;
+      fees = data?.buyerFee;
+    } else if (data.aggressor === Schema.Side.SIDE_SELL) {
+      role = MAKER;
+      fees = data?.sellerFee;
+    } else {
+      role = '-';
+      fees = !isEmptyFeeObj(data?.buyerFee) ? data.buyerFee : data.sellerFee;
+    }
+  } else if (data?.seller.id === partyId) {
+    if (data.aggressor === Schema.Side.SIDE_SELL) {
+      role = TAKER;
+      fees = data?.sellerFee;
+    } else if (data.aggressor === Schema.Side.SIDE_BUY) {
+      role = MAKER;
+      fees = data?.buyerFee;
+    } else {
+      role = '-';
+      fees = !isEmptyFeeObj(data.sellerFee) ? data.sellerFee : data.buyerFee;
+    }
+  } else {
+    return { role: '-', fees: undefined };
+  }
+
+  // We make the assumption that the market state is active if the maker fee is zero on both sides
+  // This needs to be updated when we have a way to get the correct market state when that fill happened from the API
+  // because the maker fee factor can be set to 0 via governance
+  const marketState =
+    data?.buyerFee.makerFee === data.sellerFee.makerFee &&
+    new BigNumber(data?.buyerFee.makerFee).isZero()
+      ? Schema.MarketState.STATE_SUSPENDED
+      : Schema.MarketState.STATE_ACTIVE;
+  return { role, fees, marketState };
+};
+
+export const getFeesBreakdown = (
+  role: Role,
+  fees: TradeFeeFieldsFragment,
+  marketState: Schema.MarketState = Schema.MarketState.STATE_ACTIVE
+) => {
+  // If market is in auction we assume maker fee is zero
+  const isMarketActive = marketState === Schema.MarketState.STATE_ACTIVE;
+
+  // If role is taker, then these are the fees to be paid
+  let { makerFee, infrastructureFee, liquidityFee } = fees;
+  // If role is taker, then these are the fees discounts to be applied
+  let {
+    makerFeeVolumeDiscount,
+    makerFeeReferralDiscount,
+    infrastructureFeeReferralDiscount,
+    infrastructureFeeVolumeDiscount,
+    liquidityFeeReferralDiscount,
+    liquidityFeeVolumeDiscount,
+  } = fees;
+
+  if (isMarketActive) {
```

Comment:
Mentioned in the other PR but we should make sure this is covered in the unit tests for this function